src/signature.rs

```diff
@@ -9,33 +9,23 @@
 
 //! An ed25519 signature.
 
-use core::fmt::Debug;
+#![allow(non_snake_case)]
+#![allow(absolute_paths_not_starting_with_crate)]
 
 use curve25519_dalek::edwards::CompressedEdwardsY;
 use curve25519_dalek::scalar::Scalar;
 
-#[cfg(feature = "serde")]
-use serde::de::Error as SerdeError;
-#[cfg(feature = "serde")]
-use serde::de::Visitor;
-#[cfg(feature = "serde")]
-use serde::{Deserialize, Serialize};
-#[cfg(feature = "serde")]
-use serde::{Deserializer, Serializer};
-
 use crate::constants::*;
-use crate::errors::*;
 
-/// An ed25519 signature.
-///
-/// # Note
-///
-/// These signatures, unlike the ed25519 signature reference implementation, are
-/// "detached"—that is, they do **not** include a copy of the message which has
-/// been signed.
-#[allow(non_snake_case)]
-#[derive(Copy, Eq, PartialEq)]
-pub struct Signature {
+// Re-export the `ed25519` crate's Signature type
+pub use ed25519_crate::{Signature, signature::Signature as SignatureTrait};
```

Comment:
I really dislike naming a trait `*Trait`, but it was necessary here to avoid a naming collision. Fortunately importing it is rarely needed: the only methods the itself trait provides are `from_bytes` and `as_slice`, with equivalent functionality available through `TryFrom<&[u8]>` and `AsRef<[u8]>` impls.

Otherwise its main function is to provide a set of other trait bounds.
Ideally I think the `signature` module could be eliminated and the `signature` crate publicly re-exported at the toplevel.

Reply:
Put the trait in a `src/signature/prelude.rs` maybe? Although I personally find that "prelude" naming pretentious, and have "traits" submodules usually.